src/writing-your-own-application/generate-project/esp-template.md

```diff
@@ -0,0 +1,210 @@
+# Understanding esp-template
+
+Now that we know how to [generate a no_std project], let's inspect what the generated
+project contains and try to understand every part of it.
+
+## Inspecting the generated Project
+
+When creating a project from [esp-template] with the following answers:
+-  Which MCU to target? · `esp32c3`
+-  Use template default values? · `true`
+
+For this explanation we will use the default values, if you want further modifications, see the [addtional prompts] when not using default values.
+
+
+It should generate a file structure like this:
+
+```text
+├── .cargo
+│   └── config.toml
+├── src
+│   └── main.rs
+├── .gitignore
+├── Cargo.toml
+├── LICENSE-APACHE
+├── LICENSE-MIT
+└── rust-toolchain.toml
+```
+
+Before going further let's see what these files are for.
+
+- [.cargo/config.toml]
+    - The Cargo configuration
+    - This defines a few options to correctly build the project
+    - Contains `runner = "espflash flash --monitor"` - this means you can just use `cargo run` to flash and monitor your code
+- src/main.rs
+    - The main source file of the newly created project
+    - We will examine its content in the next section
```

Comment:
Following the pattern for other files, suggest either marking `src/main.rs` as a link to the main.rs section below or adding a bullet point: For details, see the main.rs section below.

The same is applicable to the _Understanding esp-idf template_ section.